gui/src/main.rs

```diff
@@ -70,25 +60,62 @@ impl Default for GuiConfig {
 }
 
 pub struct Gui {
+    pub api: PostieApi,
     pub config: Arc<RwLock<GuiConfig>>,
     pub response: Arc<RwLock<Option<Value>>>,
+    pub headers: Rc<RefCell<Vec<(bool, String, String)>>>,
+    pub environment: Rc<RefCell<api::domain::environment::EnvironmentFile>>,
+    pub env_vars: Rc<RefCell<Vec<(bool, String, String, String)>>>,
 }
 impl Default for Gui {
     fn default() -> Self {
         Self {
+            api: PostieApi::new(),
             config: Arc::new(RwLock::new(GuiConfig::default())),
             response: Arc::new(RwLock::new(None)),
+            headers: Rc::new(RefCell::new(vec![
+                (
+                    true,
+                    String::from("Content-Type"),
+                    String::from("application/json"),
+                ),
+                (true, String::from("User-Agent"), String::from("postie")),
+                (
+                    true,
+                    String::from("Cache-Control"),
+                    String::from("no-cache"),
+                ),
+            ])),
+            environment: Rc::new(RefCell::new(EnvironmentFile {
```

Comment:
This default `environment` is currently unused, since I couldn't get it to play nice with the table below. Kept here for now but `env_vars` is the thing actually used. For the time being we may just have to convert a saved db model to a vector of those tuples?